src/components/Bag/index.js

```diff
@@ -0,0 +1,540 @@
+import React, { useState, useEffect, useCallback } from "react";
+
+import {
+    Badge,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    IconButton,
+    List,
+    ListItem,
+    ListItemAvatar,
+    Avatar,
+    ListItemText,
+    ListItemSecondaryAction,
+    makeStyles,
+    MenuItem,
+    Button,
+    useMediaQuery,
+    Select,
+    Typography,
+    FormControl,
+    InputLabel,
+} from "@material-ui/core";
+import {
+    Delete,
+    GetApp,
+    People,
+    Clear,
+    ClearAll,
+    Close,
+    ShoppingBasket as ShoppingBasketIcon,
+} from "@material-ui/icons";
+
+import SharingView from "components/sharing";
+
+import { build as buildID } from "@cyverse-de/ui-lib";
+
+import constants from "./constants";
+
+import withErrorAnnouncer from "../utils/error/withErrorAnnouncer";
+import DownloadLinksDialog from "./downloads";
+
+import * as facade from "../../serviceFacades/bags";
+import { Skeleton } from "@material-ui/lab";
+import { TYPE as SHARING_TYPE } from "components/sharing/util";
+
+import {
+    createNewBagItem,
+    FILE_TYPE,
+    FOLDER_TYPE,
+    ANALYSIS_TYPE,
+    APP_TYPE,
+} from "./classes";
+import { useTranslation } from "i18n";
+import { useTheme } from "@material-ui/styles";
+import { useUserProfile } from "contexts/userProfile";
+
+const useStyles = makeStyles((theme) => ({
+    help: {
+        paddingTop: theme.spacing(2),
+        paddingBottom: theme.spacing(1),
+    },
+    paper: {
+        width: theme.spacing(60),
+        height: theme.spacing(70),
+    },
+    closeButton: {
+        position: "absolute",
+        right: theme.spacing(1),
+        top: theme.spacing(1),
+    },
+    button: {
+        margin: theme.spacing(2),
+        width: theme.spacing(20),
+
+        [theme.breakpoints.down("sm")]: {
+            // width: "100%",
+            margin: theme.spacing(1),
+            width: theme.spacing(20),
+        },
+    },
+    actionContainer: {
+        display: "flex",
+        flexFlow: "row wrap",
+        justifyContent: "flex-end",
+    },
+    formControl: {
+        minWidth: theme.spacing(20),
+    },
+    itemText: {
+        textOverflow: "ellipsis",
+        overflow: "hidden",
+    },
+}));
+
+const BagSkeleton = () => (
+    <Skeleton variant="rect" animation="wave" height={100} width="100%" />
```

Comment:
A static ID on the loading skeletons is usually pretty helpful for testing

Reply:
Oops, yup. I always forget that.

Reply:
Working on a fix in a separate PR.